2-mandatory/4-stocks.js

```diff
@@ -63,24 +68,38 @@ function getPriceChanges(closingPricesForAllStocks) {
     The stock ticker should be capitalised.
     The price should be shown with exactly 2 decimal places.
 */
-function highestPriceDescriptions(closingPricesForAllStocks, stocks) {
-    // TODO
-}
 
+function highestPriceDescriptions(closingPricesForAllStocks, stocks)
+{
+    let HighestPriceText = [];
 
+    for(let i = 0; i < closingPricesForAllStocks.length; i++)
+    {
+        let AllHighestPrices = 0;
+        for(let i2 = 0; i2 < closingPricesForAllStocks[i].length; i2++)
+        {
+            if(closingPricesForAllStocks[i][i2] > AllHighestPrices)
+            {
+                AllHighestPrices = closingPricesForAllStocks[i][i2];
+            }
+        }
+        
+        HighestPriceText.push("The highest price of " + STOCKS[i].toUpperCase() + " in the last 5 days was " + AllHighestPrices.toFixed(2));
+    }
+
+    return HighestPriceText;
+}
```

Comment:
Great job! Just one minor detail, it may be better remove commented out code in order to find the relevant code easier.